pkg/http/server.go

```diff
@@ -149,12 +149,23 @@ func (srv *Server) Shutdown() error {
 	}
 }
 
+// Secure Go implementations of TLS ciphers
+var defaultCipherSuites = []uint16{
+	tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+	tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+}
+
 // NewServer - creates new HTTP server using given arguments.
 func NewServer(addrs []string, handler http.Handler, certificate *tls.Certificate) *Server {
 	var tlsConfig *tls.Config
 	if certificate != nil {
 		tlsConfig = &tls.Config{
 			PreferServerCipherSuites: true,
+			CipherSuites:             defaultCipherSuites,
```

Comment:
could you add a unit tests to address below cases for default http server and TLSConfig'ed http server?
1. http client connecting with `defaultCipherSuites`
2. http client connecting without `defaultCipherSuites`
3. http client connecting with unsupported cipher

Reply:
Added tests - PTAL